lib/containers/utils.pm

```diff
@@ -70,12 +70,24 @@ sub basic_container_tests {
     #   - pull image of openSUSE Tumbleweed
     assert_script_run("$runtime image pull $tumbleweed", timeout => 600);
 
+    # All images can be listed
+    assert_script_run("$runtime image ls");
     # Local images can be listed
     assert_script_run("$runtime image ls none");
-    #   - filter with tag
-    assert_script_run(qq{$runtime image ls $alpine | grep "alpine\\s*$alpine_image_version"});
-    #   - filter without tag
-    assert_script_run(qq{$runtime image ls $hello_world | grep "hello-world\\s*latest"});
+    # Images from docker.io registry are listed without the 'docker.io/library/'
+    # Images from custom registry are listed with the 'server/library/'
+    # We should filter images the same way they are listed.
+    if ($registry =~ /docker\.io/) {
+        #   - filter with tag
+        assert_script_run(qq{$runtime image ls alpine:$alpine_image_version | grep "alpine\\s*$alpine_image_version"});
+        #   - filter without tag
+        assert_script_run(qq{$runtime image ls hello-world | grep "hello-world\\s*latest"});
```

Comment:
i believe that if you actually modify the assignments at the beginning of the script, it would be more clear and then you will not need this if..else.
ex: `my $alpine               = is_private_registry ? "$registry/library/alpine:$alpine_image_version"; "alpine:$alpine_image_version"`
wdyt?

Reply:
Should be done. Doing another round of VRs now.